src/lib/metrics.py

```diff
@@ -57,16 +68,36 @@ def to_string(self) -> str:
 class Dimension:
     """Represents singular dimension."""
 
+    id: Text
+    value: Text
     dimension: Text
     source: Text
+    key_for_get_func_create_entity_id: Text
+    key_for_create_entity_id: Text
+    key_for_fetch_metric: Text
+    key_for_send_to_dynatrace: Text
 
     def __init__(self, **kwargs):
```

Comment:
This class, and especially its constructor seems overly complicated compared to how much information it actually carries. I see you replaced all usages of `dimension` and `source` in code except for L43 in `ids.py`. Maybe you should also replace it there and get rid of the `dimension` and `source` fields altogether and just use your newly introduced `key_for_*` fields?

Aside from their lack of usage in the rest of the codebase, these `dimension` and `source` fields introduce an unnecessary layer of indirection in the constructor when it comes to calculating the values of `key_for_*` fields. I think this constructor could be made simpler if we removed them

Reply:
Cleaned this up a bit, tests seem fine so far. It should look better than what you've commented on here, not 100% sure if better than what was here originally - so opinions welcome

Reply:
Looks much better now! I'm not checking if all the values stayed the same after the revamp but I trust that if you have tests for it then we're good